torch/csrc/autograd/autograd_meta.cpp

```diff
@@ -85,6 +85,9 @@ namespace {
         return false;
       }
     }
+    if (!at::_has_same_storage_numel(base, other)) {
```

Comment:
This function used to be "cheap" as it was never going through a full dispatch. Is there a measurable impact to this call?

Reply:
Benchmark to compare:
```
Before: 2.26 us, 1638366 instr
After : 2.44 us, 1686272 ins
```

<details>
<summary>
code
</summary>

```python
setup="""
import torch.autograd.forward_ad as fwAD
import torch

primal = torch.rand(10, 10)
tangent = torch.rand(10, 10)
try:
    torch._C._exit_dual_level(level=0)
except:
    pass
torch._C._enter_dual_level()
"""

stmt="""
torch._VF._make_dual(primal, tangent, level=0)
"""

timer = Timer(stmt, setup)

print(timer.timeit(10000))
print(timer.collect_callgrind(100))
```

</details>

From measuring `make_dual` there actually seems to be a measurable difference of around ~300 ns (~0.3 us) overhead, which seems pretty expected for dispatcher from what I remember from benchmarking the autograd-not-implemented kernel.

The relative slowdown is probably even larger if we call directly into `_set_fw_grad`, but just to imagine what the impact of this would be, if a model computes _set_fw_grad 1000 times, it would take 0.3ms longer to run.

Reply:
Sounds good!